### DISCUSSION TITLE
deploy contract to main net

### DISCUSSION THREAD

**Discussion by ghost** [hashlips-lab/nft-erc721-collection#213] [Q&A] [answered]
Hi everyone, 

I'm a little confused atm hoping someone can help.

I have finished my contract and tested on rinkeby. Firstly I have created a few contracts trying to get things correct, therefore when viewing on opensea testnet I have created multiple version of my collection name ie; v1, v2, etc. I'm guessing this would not have any bearing to the main net collection name. I also assume that revealing the metadata on the testnet also has no bearing? I just want to make sure before committing.

The other thing I wanted to check was how to actually deploy, I have been using truffle dashboard to use my dapp, do I simply change my wallet to main net or do I need to somehow change the contract address?

Would really appreciate some help!

Thanks

**Comment by GuyoSupreme** [comment, marked as answer]
When you deploy to mainnet, the collection name can be whatever you want - it doesn't matter what your collection names are on testnet.  Also, the metadata on testnet does not "have any bearing", BUT if somebody views that metadata on testnet and you keep the same metadata for mainnet, then they basically have a preview of that metadata for mainnet. I hope that makes sense.

As far as deploying to an actual web server, I also have this question so I am hoping that it is answered soon. I hope somebody who has already done this can comment.

**Reply by ghost** [reply]
Thanks for the reply, I thought this was the case but as I'm sure you have I've done so much work just don't want to mess up at the last stage! I'm pretty sure that for deploying the contract all we need to do change wallet to mainnet but I don't wanna say 100% and mess you up. If I find the answer I'll let you know! Thanks & good luck with your collection

**Reply by GuyoSupreme** [reply]
Oops, my mistake. We are actually talking about 2 different things in terms of "deploying". You are talking about deploying the Smart Contract to mainnet and I am talking about deploying the Minting Dapp to a web server.

I am also pretty sure that all you need to do in your case is change the CollectionConfig.ts file to your mainnet values. But I am also not 100% sure.

**Comment by progressaded** [comment]
Hi @cryptocraig-nft , I am having the same quetsion as you  . Did you found out if its enough to just switch the wallet adress ?
